File: RepositoryPattern/SampleApplicationForGM/app/src/main/java/com/example/sampleapplicationforgm/event/CustomEvent.java
package com.example.sampleapplicationforgm.event;

public class CustomEvent<T>{

    private boolean eventHasBeenHandled = false;
    private T content;

    public CustomEvent(T content) {
        this.content = content;
    }

    public boolean isEventHasBeenHandled() {
        return eventHasBeenHandled;
    }

    public T getContentIfEventNotHandled() {
        if (eventHasBeenHandled) {
            return null;
        } else {
            eventHasBeenHandled = true;
            return content;
        }
    }
}

File: RepositoryPattern/SampleApplicationForGM/app/src/main/java/com/example/sampleapplicationforgm/event/CustomEvent.kt
package com.example.sampleapplicationforgm.event

open class CustomEvent<T>(private val content: T) {
    var isEventHasBeenHandled = false
        private set

    val contentIfEventNotHandled: T?
        get() = if (isEventHasBeenHandled) {
            null
        } else {
            isEventHasBeenHandled = true
            content
        }

}